src/baguetteBox.js

```diff
@@ -148,50 +146,75 @@
         // For each gallery bind a click event to every image inside it
-        var gallery = document.querySelectorAll(selector);
-        galleries.push(gallery);
-        [].forEach.call(gallery, function(galleryElement) {
+        var galleryNodeList = document.querySelectorAll(selector),
+            selectorData = {
+                galleries: [],
+                nodeList: galleryNodeList
+            };
+        data[selector] = selectorData;
+        
+        [].forEach.call(galleryNodeList, function(galleryElement) {
             if (userOptions && userOptions.filter) {
                 regex = userOptions.filter;
             }
             // Filter 'a' elements from those not linking to images
-            var tags = galleryElement.getElementsByTagName('a');
-            tags = [].filter.call(tags, function(element) {
+            var tagsNodeList = galleryElement.getElementsByTagName('a');
+            tagsNodeList = [].filter.call(tagsNodeList, function(element) {
                 return regex.test(element.href);
             });
+            var gallery = [];
+            
+            if (tagsNodeList.length === 0) {
+                return;
+            }
+            selectorData.galleries.push(gallery);
 
-            // Get all gallery images and save them in imagesMap with custom options
-            var galleryID = imagesMap.length;
-            imagesMap.push(tags);
-            imagesMap[galleryID].options = userOptions;
-
-            [].forEach.call(imagesMap[galleryID], function(imageElement, imageIndex) {
+            [].forEach.call(tagsNodeList, function(imageElement, imageIndex) {
                 var imageElementClickHandler = function(event) {
                     event.preventDefault ? event.preventDefault() : event.returnValue = false; // jshint ignore:line
-                    prepareOverlay(galleryID);
+                    prepareOverlay(gallery, userOptions);
                     showOverlay(imageIndex);
                 };
-                imagedEventHandlers[galleryID + '_' + imageElement] = imageElementClickHandler;
+                var imageItem = {
+                    eventHandler: imageElementClickHandler,
+                    imageElement: imageElement
+                };
                 bind(imageElement, 'click', imageElementClickHandler);
+                gallery.push(imageItem);
             });
         });
     }
 
     function unbindImageClickListeners() {
-        galleries.forEach(function(gallery) {
-            [].forEach.call(gallery, function() {
-                var galleryID = imagesMap.length - 1;
-                [].forEach.call(imagesMap[galleryID], function(imageElement) {
-                    unbind(imageElement, 'click', imagedEventHandlers[galleryID + '_' + imageElement]);
-                });
-                imagesMap.pop();
+        for (var selector in data) {
+            if (data.hasOwnProperty(selector)) {
+                destroyGalleriesForSelector(selector);
```

Comment:
I have renamed destroyGalleriesForSelector() to removeFromCache() - because this function unbinds the listeners and removes the selector from the data.
I have also renamed unbindImageClickListeners() to clearCachedData(). 
Tell me if you have a better idea and I will rename them :)